common/src/main/java/org/valkyrienskies/mod/mixin/feature/teleport_reconnected_player_to_ship/MixinServerPlayer.java

```diff
@@ -0,0 +1,89 @@
+package org.valkyrienskies.mod.mixin.feature.teleport_reconnected_player_to_ship;
+
+import com.mojang.authlib.GameProfile;
+import javax.annotation.Nullable;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import org.joml.Vector3d;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.valkyrienskies.core.api.ships.Ship;
+import org.valkyrienskies.mod.common.VSGameUtilsKt;
+import org.valkyrienskies.mod.common.util.EntityDraggingInformation;
+import org.valkyrienskies.mod.common.util.IEntityDraggingInformationProvider;
+
+@Mixin(ServerPlayer.class)
+public abstract class MixinServerPlayer extends Player {
+
+    @Shadow
+    public abstract ServerLevel serverLevel();
+
+    // The maximum number of ticks before the player is no longer considered aboard a ship
+    // if they are no longer directly touching it
+    @Unique
+    private static final int MAX_ONBOARD_TICKS = 4;

```

Comment:
Could you move this to the config class, so that users themselves can modify this.

Reply:
No problem!!

Fixed in 266c308496ecd02b4417c286ead432f648b07bcb